src/Integration/Binding/SolutionBindingOperation.cs

```diff
@@ -129,25 +121,18 @@ public void Prepare(CancellationToken token)
 
                 var info = keyValue.Value;
 
-                sourceControlledFileSystem.QueueFileWrites(info.SolutionLevelFilePaths, () =>
```

Comment:
This is the key part of the change - we're no longer queuing the changes, we're writing them directly.